test/Datadog.Trace.Tests/Util/ArrayBuilderTests.cs

```diff
@@ -0,0 +1,51 @@
+// <copyright file="ArrayBuilderTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Linq;
+using Datadog.Trace.Util;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Util
+{
+    public class ArrayBuilderTests
+    {
+        // Store the builder in a field, to make sure copy-semantics of structs don't get in the way
+        // of how the type is expected to be used
+        private ArrayBuilder<int> _builder;
+
+        [Fact]
+        public void BuildArray()
+        {
+            const int numberOfElements = 20;
+
+            _builder = default;
+
+            for (int i = 0; i < numberOfElements; i++)
+            {
+                _builder.Add(i);
+            }
+
+            _builder.Count.Should().Be(numberOfElements);
+
+            var result = _builder.GetArray();
+
+            result.Should().BeEquivalentTo(Enumerable.Range(0, numberOfElements));
+        }
+
+        [Fact]
+        public void InitialCapacity()
```

Comment:
Can you also write a test asserting how the array builder grows?

Reply:
Good call, I actually found a bug while doing this.

Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1502/commits/8c4919102d2ff6f1ad894c3639fcf28a3c4c8423